File: springMessengerAPI/src/main/kotlin/com/messenger/springMessengerAPI/models/Message.kt
package com.messenger.springMessengerAPI.models

import com.fasterxml.jackson.annotation.JsonManagedReference
import jakarta.persistence.*
import java.time.LocalDateTime
import java.util.*


@Entity
@Table(name = "Messages")
class Message(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        val id: Int = 0,

        @Column(name = "UserId")
        val userId: Int = 0,

        @Column(name = "TextMessage")
        val textMessage: String? = "",

        //Default to max, so never get a message that has no timesent
        @Column(name = "TimeSent")
        val timeSent: LocalDateTime = LocalDateTime.MAX,

        @Column(name = "UpdatedTime")
        val updatedTime: LocalDateTime? = null,

        @Column(name = "ConversationId")
        val conversationId: Int = 0,
)

File: MessengerAPI/MessengerAPI/Models/Message.cs
namespace MessengerAPI.Models
{
    public class Message
    {
        public int Id { get; set; }       
        public int UserId { get; set; }
#nullable enable
        public string? TextMessage { get; set; }
        public DateTime TimeSent { get; set; }
        public DateTime? UpdatedTime { get; set; }
#nullable disable

        public int ConversationId { get; set; }
        public Conversation Conversation { get; set; }
    }
}
